code/modules/shuttle/shuttle.dm

```diff
@@ -704,6 +745,8 @@
 			if(prearrivalTime && mode != SHUTTLE_PREARRIVAL)
 				set_mode(SHUTTLE_PREARRIVAL)
 				setTimer(prearrivalTime)
+				/// Run on_prearrival after setting the mode rather than waiting for next check, as this doesn't occur until landing is complete.
+				on_prearrival()
```

Comment:
All assumptions prior to this PR have expected on_prearrival to occur below on line 708.

As I said before, this needs to be a *new* proc e.g. `on_start_prearrival`. Alternatively you do something similar inside of `set_mode(SHUTTLE_PREARRIVAL)`

Then move all landing light handling that is `on_prearrival` to the new `on_start_prearrival`